src/main/java/Deadline.java

```diff
@@ -0,0 +1,26 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a category of tasks called Deadline, that includes the task
+ * and the deadline as to which the task has to be completed by
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
Perhaps you could consider a more descriptive variable name?

Reply:
Thank you so much for all the really useful comments!! Will fix them (: